### PR TITLE
Fix various issues

### LINKED ISSUE #186: How can i use OCO Order with ws2?
I tried 'ocoorder': true,
buy_price_oco and sell_price_oco

Where can I get an example of a OCO order?

### LINKED ISSUE #209: sendCalc not stringified before sending
This is an easy one
 _sendCalc (msg) {
    debug('req calc: %j', msg)

    this._ws.send(msg)
  }

The msg variable should be JSON.stringify before it is sent to the WS.

_sendCalc (msg) {
    debug('req calc: %j', msg)

    this._ws.send(JSON.stringify(msg));
  }

### PR DESCRIPTION
**Pull request by f3rno**
Needs https://github.com/bitfinexcom/bitfinex-api-node/pull/227

* stringifies `calc` command ws payload, closes #209 
* adds OCO example, closes #186 
* adds wrapper for RESTv1 methods within RESTv2
* adds `updateSettings`, `deleteSettings`, `getSettings` to `RESTv2`
* log statements/standard --fix/fixed tests

### CODE REVIEW

File: lib/transports/ws2.js [hunk 1]
```diff
@@ -179,7 +192,12 @@ class WSv2 extends EventEmitter {
     const { sig } = genAuthSig(this._apiSecret, authPayload)
 
     return new Promise((resolve, reject) => {
-      this._ws.once('auth', () => resolve())
+      this.once('auth', () => {
+        debug('authenticated')
+        resolve()
+      })
+
+      debug('authenticating w/ key %s', this._apiKey)
```
Comment (robertkowalski) [inline lib/transports/ws2.js]: uh, maybe not good to log access keys security-wise, they can end up in log files etc

File: lib/transports/rest2.js [hunk 2]
```diff
@@ -15,7 +15,12 @@ const {
   Trade,
   Wallet,
   Alert,
+<<<<<<< HEAD
   Tick
+=======
+  Ticker,
+  Candle
+>>>>>>> 03c086b... (refactor) improve REST2 methods that call through to REST1
```
Comment (robertkowalski) [inline lib/transports/rest2.js]: here is an unresolved conflict

File: lib/transports/ws2.js [hunk 3]
```diff
@@ -1124,7 +1124,7 @@ class WSv2 extends EventEmitter {
   _sendCalc (msg) {
     debug('req calc: %j', msg)
 
-    this._ws.send(msg)
+    this._ws.send(JSON.stringify(msg))
```
[the remark this answers is not in the document]
Comment (f3rno) [reply lib/transports/ws2.js]: I added a test in cbc4a49, good idea :)

File: lib/transports/ws2.js (hunk 1 above)
Comment (f3rno) [reply lib/transports/ws2.js]: Removed

File: lib/transports/rest2.js (hunk 2 above)
Comment (f3rno) [reply lib/transports/rest2.js]: Fixed, it was a remnant of awaiting the rebase I think.

**Pull request merged**